### ISSUE TITLE
error Connection lost. Reconnecting

### ISSUE THREAD

**Issue by Sod-Almighty** [francois2metz/em-eventsource#12]
I'm convinced this gem is utterly broken:
```
EM.run do
	source = EventMachine::EventSource.new "http://nas:2347/sse", {}, {'Accept' => 'text/event-stream'}
	source.inactivity_timeout = 0
	source.message do |message|
		puts message
	end
	source.on "notification" do |message|
		puts "notification #{message}"
	end
	source.error do |error|
		puts "error #{error}"
	end
	source.open do
		puts "opened"
	end
	source.start # Start listening
end
```
```
opened
error Connection lost. Reconnecting.
opened
error Connection lost. Reconnecting.
opened
error Connection lost. Reconnecting.
...
```
Sinatra server output:
```
I, [2018-05-15T09:24:00.199421 #12430]  INFO -- : 10.0.0.2 - - "SSE /sse HTTP/1.1" 200 -
W, [2018-05-15T09:24:00.200600 #12430]  WARN -- : closed stream
I, [2018-05-15T09:24:03.205054 #12430]  INFO -- : 10.0.0.2 - - "SSE /sse HTTP/1.1" 200 -
W, [2018-05-15T09:24:03.210790 #12430]  WARN -- : closed stream
I, [2018-05-15T09:24:06.211593 #12430]  INFO -- : 10.0.0.2 - - "SSE /sse HTTP/1.1" 200 -
W, [2018-05-15T09:24:06.212589 #12430]  WARN -- : closed stream
```
I've spent two hours trying to debug this, and I'm none the wiser.

**Comment by francois2metz** [comment]
Hi, could you try to remove `source.inactivity_timeout = 0`?

**Comment by Sod-Almighty** [comment]
I only added that line in an attempt to solve the problem.

**Comment by francois2metz** [comment]
Ok, could you give me a working sinatra server to reproduce it?

**Comment by Sod-Almighty** [comment]
I will open a port on my router to allow you to access the locally-hosted server. Give moment.

Incidentally, I misspoke slightly in the original report. I am actually using [Angelo](https://github.com/kenichi/angelo), rather than Sinatra.

**Comment by Sod-Almighty** [comment]
Would you like me to email you the address and port at francois@2metz.fr ?

**Comment by francois2metz** [comment]
Hi Sod, having some code to reproduce it locally would be easier for me.

**Comment by Sod-Almighty** [comment]
Oh, I see. You want some Ruby to host a server. Give me ten minutes.

**Comment by Sod-Almighty** [comment]
Alright, try this:
```
#!/usr/bin/env ruby
require 'angelo'

class HTTPServer < Angelo::Base		#I::AngeloServer
	# include Angelo::Base::DSL
	report_errors!
	def self.start!
		addr '0.0.0.0'
		port 2347
	end
	eventsource '/sse' do |s|
		s.responder.content_type :json
		sses[:notification] << s
	end
	get '/' do
		halt 200, "Nothing to see here"
	end
	def self.notify **data
		sses[:notification].event **data
	end
end

Thread.new do
	while true
		sleep 1
		begin
			HTTPServer.notify test: "test"
		rescue Exception => e
			puts e
		end
	end
end
HTTPServer.start!
HTTPServer.run!
```

**Comment by francois2metz** [comment]
With angelo, I didn't manage to make it works. With my own sse server, it works like a charm.

**Comment by Sod-Almighty** [comment]
Well, the [Linux sse command](https://github.com/radiospiel/sse) connects to Angelo fine:
```
# ~/sse/bin/sse -i http://nas:2347/sse
EVENT event:<none> (19 byte)
EVENT=notification
�<�
{"test":"test"}

EVENT event:<none> (16 byte)
EVENT=notification

{"test":"test"}

EVENT event:<none> (16 byte)
EVENT=notification

{"test":"test"}
```
Compiling the `sse` command is a pain in the bum, but I eventually managed it. And the fact that it works kinda suggests that it's your gem at fault.

**Comment by Sod-Almighty** [comment]
Also, I have used SSE successfully in Javascript on my intranet before, with an Angelo server, and _that_ works also. Sorry dude, it's not Angelo to blame here.

**Comment by francois2metz** [comment]
There is a crash on the angelo server code you gave me. Could you give me a more complete example ?

**Comment by Sod-Almighty** [comment]
That is pretty much the exact code I am using. What was the crash?

**Comment by francois2metz** [comment]
Something about report_errors. Don"t have the exact crash at this moment.

**Comment by francois2metz** [comment]
For some unknown reason, the socket is immediately closed on one side. Don't know what's wrong. It only happens with angelo for now.

**Comment by francois2metz** [comment]
Okay, I found the issue. I'll try to fix it soon.

**Comment by Sod-Almighty** [comment]
Great

**Comment by francois2metz** [comment]
The issue has been fixed and I released the version 0.2.2 on rubygems.
